ros2_batch_job/__main__.py

```diff
@@ -777,5 +782,24 @@ def _fetch_repos_file(url, filename, job):
     with open(filename, 'r') as f:
         print(f.read())
 
+# Checks if the given branch exists in at least one repo
+def check_branch_exists(filename, branch):
+    # Populate all .git repos
+    lines = []
+    with open(filename) as f: lines = f.readlines()
+
+    urls = []
+    for line in lines:
+        text = line.strip()
+        if text[-4:] == ".git":
```

Comment:
It'd be nice if there is a vcs command we could use instead. I think checking for a ".git" suffix might be too brittle since I don't think it's a requirement that it's there. Also, since the repositories are already cloned locally, it seems a bit redundant to re-parse the repos files.

Reply:
That's right, this is too brittle.  The `.git` isn't required (though it is recommended).

Actually, I wonder if it would be better to just parse these `repos` files as the YAML files they are.  That will be way less brittle, and doesn't require us to use `os.system` (or this brittle parsing) at all.